### ISSUE TITLE
Android Studio-specific issue Workaround 

### ISSUE THREAD

**Issue by skreenr** [ClaudiuGeorgiu/Obfuscapk#111]
As reported in the Bug issue #110 , Android Studio-generated APK seems to run into conflict with Obfuscapk' apktool-based workflow

How feasible would be a Hybrid workflow, where some critical part of the app would be compiled in Eclipse, then obfuscated, then imported into Android Studio to add Google-specific libraries not available in Eclipse?  - at least not "easily available" ?

Can we drop the Eclipse step and directly obfuscate the critical Java code pieces instead, before importing into AS?
What kind of command lines can be used for it ?

**Comment by ClaudiuGeorgiu** [comment]
I don't think an hybrid workflow as you described would work, and Obfuscapk cannot obfuscate directly Java code. Maybe this issue could be solved by using `--use-aapt2` flag, as you described in #110.

**Comment by skreenr** [comment]
What I was suggesting, consider leaving the pieces half-way without the final apktool build call, to give people the ability to use take the obfuscated pieces and import into A-Studio to integrate with Google libs which may be causing trouble. Or it may be the trouble is due to Google's new ways of building APKs as mentioned in apktool blog– regardless, my idea is to be able to build a minimal functionality with Eclipse, then obfuscate critical code, take that to Android Studio, add other pieces and complete the project there. See what I mean?

**Comment by skreenr** [comment]
Let me rephrase it in architecture speak: 

You are offering an end-to-end solution, which works fine in many cases, but not all.
What I am suggesting is, you open your box up so people can use not the full pipeline, but just some of it as part of their overall workflow.
Look to apktool itself: you can use it as a standalone tool, and you can integrate it into a workflow, just like obfuscapk does.
So, it increases the value of your offering when you give that flexibility. 

As a context, in my case I have build an early version in Eclipse and obfuscated it successfully.
Later I tried adding Google libs such as AndroidX and FCM, which is not easy to do using Eclipse (I still dont have idea if that's possible). I build the APK, and now trying to obfuscate it, yet getting this errors. The minimal 2M TestAstu does not have FCM, so this one may be breaking things. I can't drop FCM, instead, if I could obfuscate my original core to some intermediate format, then take it into AStudio to add FCM and build APK, that could be a workaround.

Does it make sense?

**Comment by ClaudiuGeorgiu** [comment]
> my idea is to be able to build a minimal functionality with Eclipse, then obfuscate critical code, take that to Android Studio, add other pieces and complete the project there.

To me, this seems a very unlikely use case. Why not use Android Studio from the beginning? Moreover, I have no experience with Eclipse so I can't help with this.

> What I am suggesting is, you open your box up so people can use not the full pipeline, but just some of it as part of their overall workflow.

Well, the project is open source so it's already open 😄 the pipeline is made by obfuscators and the user can choose the obfuscators he wants.

I guess you found an use case for which Obfuscapk is not the right tool, as you said it works fine in some cases but there are other situations in which Obfuscapk doesn't work/has errors: this could be one of those situations. Personally, I think (but I might be wrong) that what you are trying to do is not achievable with Obfuscapk, sorry.

**Comment by skreenr** [comment]
Let me explain: When I built the early version in Eclipse, everything was obfuscating/signing/installing and running on device fine. Now when I

**Comment by ClaudiuGeorgiu** [comment]
> How do I stop the process before going into the final rebuild and signing, so as to have importable .java files I can then bring into A Studio?

You can't. Obfuscapk doesn't work with java files. I mean, you can stop the process at the point you want by simply not using the last three obfuscators (rebuild, alignment and signature), but you will get no java file, only [smali](https://stackoverflow.com/a/30837786/5268548) files.

**Comment by skreenr** [comment]
OK I get that
What can I do with smali files?
Clearly Apktool has been successful in gainfully using those.
